syntax = "proto3";

package viamlabs.service.audioout.v1;

import "google/api/annotations.proto";

service AudiooutService {
  rpc Play(PlayRequest) returns (PlayResponse) {
    option (google.api.http) = {
      post: "/acme/api/v1/service/speech/{name}/play"
    };
  }

  rpc Stop(StopRequest) returns (StopResponse) {
    option (google.api.http) = {
      post: "/acme/api/v1/service/speech/{name}/stop"
    };
  }
}

message PlayRequest {
  string name = 1;
  string file_path = 2;
  int32 loop_count = 3;
  int32 maxtime_ms = 4;
  int32 fadein_ms =  5;
  bool block = 6;
}

message PlayResponse {
  string text = 1;
}

message StopRequest {
  string name = 1;
}

message StopResponse {
  string text = 1;
}